src/js/reducers/index.js

```diff
@@ -0,0 +1,172 @@
+import {
+  GET_PRODUCTS_REQUEST, GET_PRODUCTS_SUCCESS, GET_PRODUCTS_FAILURE,
+  GET_CONFIGS_SUCCESS, GET_CONFIGS_FAILURE,
+  LOGIN_SUCCESS, LOGIN_FAILURE,
+  GET_PRODUCT_BY_CATEGORY_REQUEST, GET_PRODUCT_BY_CATEGORY_SUCCESS, GET_PRODUCT_BY_CATEGORY_FAILURE,
+  SEARCH_REQUEST, SEARCH_SUCCESS, SEARCH_FAILURE,
+  GET_PRODUCT_BY_ID_REQUEST, GET_PRODUCT_BY_ID_SUCCESS, GET_PRODUCT_BY_ID_FAILURE,
+  ADD_ORDER_REQUEST, ADD_ORDER_SUCCESS, ADD_ORDER_FAILURE,
+  ADD_CART_REQUEST,
+  UPDATE_CART_REQUEST,
+  OPEN_MODAL_REQUEST,
+  CLOSE_MODAL_REQUEST,
+  SIGN_OUT_REQUEST,
+  SIGN_UP_REQUEST, SIGN_UP_SUCCESS, SIGN_UP_FAILURE,
+  ADD_ORDER_ITEM_REQUEST, ADD_ORDER_ITEM_SUCCESS, ADD_ORDER_ITEM_FAILURE,
+  CLOSE_ORDER_COMPLETE_MODAL_REQUEST,CLOSE_ORDER_COMPLETE_MODAL_SUCCESS,CLOSE_ORDER_COMPLETE_MODAL_FAILURE,
+  CLOSE_ORDER_SUCCESS_MODAL,
+  CLOSE_ERROR
+
+} from '../constants/action-types';
+
+
+const initialState = {
+
+  products: [],
+  product: null,
+  attributes: [],
+  attributes_values: [],
+  categories: [],
+  token: undefined,
+  login: false,
+  customer_id: undefined,
+  customer_name: undefined,
+  order_id: undefined,
+  chart: [],
+  modal: false,
+  signup: false,
+  order_state: false,
+  order_success:false,
+  error:false,
+  errorMessage:''
+};
+
+function rootReducer(state = initialState, action) {
```

Comment:
@Wajahat-Khan I suggest to split reducers to their own files, like ProductReducer, CartReducer,... and use combine reducer to sum up. In this way you will be following SRP and better state management in Store.

Reply:
Okay